tests/API/Management/UsersTest.php

```diff
@@ -4,61 +4,205 @@
 use Auth0\SDK\API\Management;
 use Auth0\Tests\API\BasicCrudTest;
 
-class UsersTest extends BasicCrudTest {
+/**
+ * Class UsersTest.
+ *
+ * @package Auth0\Tests\API\Management
+ */
+class UsersTest extends BasicCrudTest
+{
+    /**
+     * Unique identifier name for Users.
+     *
+     * @var string
+     */
+    protected $id_name = 'user_id';
 
-    protected $email;
+    /**
+     * Random number used for unique testing names.
+     *
+     * @var integer
+     */
+    protected $rand;
+
+    /**
+     * Should the results returned by getAll be searched for the created user?
+     *
+     * @var bool
+     */
     protected $findCreatedItem = false;
 
-    protected function getId($entity) {
-        return $entity['user_id'];
+    /**
+     * Connection to create the user in.
+     *
+     * @var string
+     */
+    protected $connection = 'Username-Password-Authentication';
+
+    /**
+     * UsersTest constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->rand = rand();
     }
 
-    protected function getApiClient($scopes = null) {
+    /**
+     * Return the Users API to test.
+     *
+     * @return Management\Users
+     */
+    protected function getApiClient()
+    {
+        $token = $this->getToken($this->env, ['users' => ['actions' => ['create', 'read', 'delete', 'update']]]);
+        $api = new Management($token, $this->domain);
+        return $api->users;
+    }
 
-        if ($scopes === null) {
-            $scopes = ['create', 'read', 'delete', 'update'];
-        }
+    /**
+     * Get the User create data to send with the test create call.
+     *
+     * @return array
+     */
+    protected function getCreateBody()
+    {
+        return [
+            'connection' => $this->connection,
+            'email' => 'test-create-user-' . $this->rand . '@auth0.com',
+            'password' => 'Y6t82hQjpXCMd3oD7Zsc',
+            'picture' => 'https://cdn.auth0.com/styleguide/components/1.0.8/media/logos/img/badge.png',
+            'email_verified' => false,
+            'user_metadata' => [
+                'key1' => 'value1',
+                'key2' => 'value2',
+            ],
+        ];
+    }
 
-        $env = $this->getEnv();
-        $token = $this->getToken($env, [
-            'users' => [
-                'actions' => $scopes
-            ]
-        ]);
+    /**
+     * Tests the \Auth0\SDK\API\Management\Users::getAll() method.
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     */
+    protected function getAllEntities()
+    {
+        $fields = array_keys($this->getCreateBody());
+        $fields[] = $this->id_name;
+        $page_num = 1;
 
-        $this->domain = $env['DOMAIN'];
+        // Get the second page of Users with 1 per page (second result).
+        $paged_results = $this->api->getAll([
+            'sort' => 'created_at:1'
```

Comment:
do the rest of the entities do the same sorting?

Reply:
No. I added this because we're likely to get back a lot of results and this will ensure the pagination check works regardless of the number. Other endpoints don't appear to support sorting.